### PR TITLE
Add build types to issue template

### PR DESCRIPTION
**Pull request by tjzel**
🚀

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -120,10 +120,19 @@ body:
         - Fabric (New Architecture)
 
   - type: dropdown
-    id: build-type
+    id: bundle-build-type
     attributes:
-      label: Build type
-      description: What mode your application is running?
+      label: JavaScript bundle build type
+      description: What is the type of your JavaScript bundle?
```
Comment (kacperkapusciak) [inline .github/ISSUE_TEMPLATE/bug-report.yml:126]: People will have no idea what we are asking about, we need to word it differently

Comment (tjzel) [reply .github/ISSUE_TEMPLATE/bug-report.yml:126]: I didn't come up with how to phrase it better but keep in mind that it's an optional field.

Comment (tomekzaw) [reply .github/ISSUE_TEMPLATE/bug-report.yml:126]: The official term is "bundle mode".

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -120,10 +120,19 @@ body:
...
+      options:
+        - Debug mode
+        - Release mode
```
Comment (tomekzaw) [inline .github/ISSUE_TEMPLATE/bug-report.yml:129]:
```suggestion
        - Production mode
```

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 3]
```diff
@@ -120,10 +120,19 @@ body:
...
+
+  - type: dropdown
+    id: app-build-type
+    attributes:
+      label: App build type
```
Comment (tomekzaw) [inline .github/ISSUE_TEMPLATE/bug-report.yml:134]:
```suggestion
      label: Native app build configuration/variant
```

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 4]
```diff
@@ -120,10 +120,19 @@ body:
...
-      label: Build type
-      description: What mode your application is running?
+      label: JavaScript bundle build type
+      description: What is the type of your JavaScript bundle?
+      options:
+        - Debug mode
+        - Release mode
+
+  - type: dropdown
+    id: app-build-type
```
Comment (tomekzaw) [inline .github/ISSUE_TEMPLATE/bug-report.yml:132]:
```suggestion
    id: build-mode
```

**Review by tomekzaw — CHANGES_REQUESTED** [review]
Instead of having two separate questions, wouldn't it be better have one question but with more answers? This way we can detect exotic use cases. For example:

**Build type**
- Debug app & dev bundle
- Release app & production bundle
- Release app & dev bundle (e.g. in Expo Go)
- Debug app & production bundle (e.g. when profiling JS bundle)
- Custom (e.g. E2E, other mix of debug and release)

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 5]
```diff
@@ -123,10 +123,13 @@ body:
...
       label: Build type
-      description: What mode your application is running?
+      description: What is the build configuration/variant of your native app and JavaScript bundle mode?
       options:
-        - Debug mode
-        - Release mode
+        - Debug app && Dev bundle
+        - Release app && Production bundle
+        - Debug app && Production bundle
+        - Release app && Dev bundle
```
Comment (tomekzaw) [inline .github/ISSUE_TEMPLATE/bug-report.yml:131]:
```suggestion
        - Debug app & dev bundle
        - Release app & production bundle
        - Debug app & production bundle
        - Release app & dev bundle
```

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**